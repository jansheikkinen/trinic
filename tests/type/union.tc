// Union

function main(void) -> void do
  let a: union TestUnion, b: union TestUnionB(int8, int16) = undefined;
end
