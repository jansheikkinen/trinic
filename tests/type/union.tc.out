
### LEXER ###
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:          LET,     (null), (004, 005)
009:   IDENTIFIER,          a, (004, 007)
010:            :,     (null), (004, 008)
011:        UNION,     (null), (004, 014)
012:   IDENTIFIER,  TestUnion, (004, 024)
013:            ,,     (null), (004, 025)
014:   IDENTIFIER,          b, (004, 027)
015:            :,     (null), (004, 028)
016:        UNION,     (null), (004, 034)
017:   IDENTIFIER, TestUnionB, (004, 045)
018:            (,     (null), (004, 046)
019:         INT8,     (null), (004, 050)
020:            ,,     (null), (004, 051)
021:        INT16,     (null), (004, 057)
022:            ),     (null), (004, 058)
023:            =,     (null), (004, 060)
024:    UNDEFINED,     (null), (004, 070)
025:            ;,     (null), (004, 071)
026:          END,     (null), (005, 003)

### AST GENERATION ###
Declaration FUNCTION  main
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg a
        Type UNION TestUnion
      Identifier_Arg b
        Type UNION TestUnionB
          Arg SUMARGTYPE
            Sum_Arg_Type 
              Type TYPE INT8
            Sum_Arg_Type 
              Type TYPE INT16
    Arg EXPR
      Expr UNDEFINED
